File: golang/WordsSearchResult.go
package ToolGood_Words

type WordsSearchResult struct {
	Success bool
	Start int
	End int
	Keyword string
	Index int
}

func NewWordsSearchResult(keyword string,start int,end int,index int) *WordsSearchResult  {
	return &WordsSearchResult{
		Success:true,
		Start:start,
		End:end,
		Keyword:keyword,
		Index:index,
	}
}

File: csharp/ToolGood.Words.Pinyin/internals/WordsSearchResult.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToolGood.Words.Pinyin.internals
{
    public class WordsSearchResult
    {
        internal WordsSearchResult(  int start, int end, int index,int length)
        {
            End = end;
            Start = start;
            Index = index;
            Length = length;
        }

        /// <summary>
        /// 开始位置
        /// </summary>
        public int Start { get; private set; }

        /// <summary>
        /// 结束位置
        /// </summary>
        public int End { get; private set; }

 

        /// <summary>
        /// 索引
        /// </summary>
        public int Index { get; private set; }

        public int Length { get; private set; }



    }

}
